addons/web/static/src/js/fields/special_fields.js

```diff
@@ -192,9 +196,84 @@ var FieldReportLayout = relational_fields.FieldMany2One.extend({
 });
 
 
+const IframeWrapper = AbstractField.extend({
+    description: _lt("Wrap raw html within an iframe"),
+
+    // If HTML, don't forget to adjust the sanitize options to avoid stripping most of the metadata
+    supportedFieldTypes: ['text', 'html'],
+
+    /**
+     * Note:
+     * In case of change of behaviour for this widget, please note some styles for the BaseDocumentPreview is
+     * stored here: addons/web/static/src/scss/base_document_layout.scss and contains some style linked to this
+     * widget. Proceed with caution not to break the document layout preview.
+     */
+
+    _render: function () {
+
+        this.$el.empty();
+        this.$el.attr('id', 'o_preview_iframe_wrapper')
+
+        const $spinner = $('<i class="fa fa-spinner fa-spin fa-4x" id="o_iframe_wrapper_spinner" style="color: #875a7b; z-index: 2"></i>')[0]
+        this.$el[0].appendChild($spinner)
+
+        const html_code = this.value
+
+        const iframe = document.createElement('iframe');
+        iframe.style.display = 'none'
+        iframe.setAttribute('id', 'o_preview_iframe')
+        this.$el[0].appendChild(iframe);
+
+        // Promise for tests
+        let resolver;
+        $(iframe).data('ready', new Promise(function (resolve) {
+            resolver = resolve
+        }));
+
+
+        /**
+         * Certain browser don't trigger onload events of iframe for particular cases.
+         * In our case, chrome and safari could be problematic (yet not on all machines).
+         * This rather unorthodox solution replace the onload event handler.
```

Comment:
What do you mean by "yet not on all machines"? Is this more than a browser compatiblity problem? Currently the JS Framework supports all evergreen browsers (mainly if not only: Chrome & Chromium, Opera, Safari, Firefox and Edge + their mobile equivalents).

Now if it is machine-specific I would like to know the details because I think this is a rather huge trade-off.

Reply:
So for what I know: It works on chrome on my mac. It doesn't on safari. When Nathan tried it on his laptop with chrome, it didn't worked. That's why I said "not on all machines". The truth is probably different but those iframes are touchy. Also, from the documentation and comments on iframe.onload, it is quite unreliable. It also can sometimes draw the report once, meaning the onload was properly fired, but won't be fired anymore for changes. From looking on the internet, depending on the browser, there are also different behaviors that can fire the event twice. And finally, still depending on the browser, it will behave differently depending on the attributes of the iframe. So I got a bit overwhelmed, tried a lot of things but that's currently my safest solution.

Reply:
@Arcasias

Reply:
Fair enough, let's keep it that way then.